### PR TITLE
schedule build job

### PR DESCRIPTION
**Pull request by robertylewis**
The scheduled job won't work until https://github.com/leanprover-community/mathlib/pull/3583 is merged.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -1,7 +1,10 @@
 name: build and deploy docs
 
-on: push
-
+on:
+  schedule:
+    - cron: "0 */2 * * *" # every 2 hours
+  push:
```
Comment (gebner) [inline .github/workflows/build.yml:6]:
```suggestion
  push:
    branches:
  	  - master
```

**Comment by robertylewis** [comment]
Yes, this looks better. This now moves the script from mathlib to this repo.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -1,7 +1,12 @@
 name: build and deploy docs
 
-on: push
-
+on:
+  schedule:
+    - cron: '0 */2 * * *' # every 2 hours
+  push:
+    branches:
+      - master
```
Comment (gebner) [inline .github/workflows/build.yml:8]:
```suggestion
  pull_request:
```

Sorry for going back and forth on this!

Now that the script is included in this PR, I've noticed that this check is already in the deploy script.  It's probably useful to generate the documentation as part of CI here.  Then you notice if you break something in a PR to `doc-gen`.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -35,7 +38,7 @@ jobs:
           leanproject up
 
       - name: generate docs
-        run: mathlib/scripts/deploy_docs.sh "mathlib" ".." "mathlib"
+        run: deploy_docs.sh "mathlib" ".." "mathlib"
```
Comment (gebner) [inline .github/workflows/build.yml:41]:
```suggestion
        run: ./deploy_docs.sh "mathlib" ".." "mathlib"
```

And CI has already found an error!

**Pull request merged**